lib/cocoapods-core/specification.rb

```diff
@@ -43,7 +43,7 @@ def initialize(parent = nil, name = nil, test_specification = false)
       @consumers = {}
       @parent = parent
       @hash_value = nil
-      @test_specification = test_specification
+      attributes_hash['test_specification'] = test_specification
```

Comment:
previously this was not stored in the serialized form, causing specs to be loaded from disk to be marked as non-test-specifications even though initially they were.